### PR TITLE
Fcfs testing

### PR DESCRIPTION
**Pull request by YVNGO**
leave a comment if you find a bug (_|_)

### CODE REVIEW

**Review by tejanidamni — APPROVED** [review]

File: CS3343-Project/src/Project/IntervalPair.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package Project;
+
+public class IntervalPair {
+	private int start;
+	private int second;
+	
+	IntervalPair(int _start, int _second){
+		this.start = _start;
+		this.second = _second;
```
Comment (alibimm) [inline CS3343-Project/src/Project/IntervalPair.java]: Start and end?

File: CS3343-Project/src/Project/FCFS.java [hunk 2]
```diff
@@ -46,16 +56,16 @@ public Result schedule(ArrayList<Process> processes){
...
                 Process cur_process = ready_queue.get(0); // always dispatch the first process in ready queue
-                cur_process_id = cur_process.process_id;
+                cur_process_id = cur_process.getId();
                 if (cur_process_id != prev_process_id)
                 { // store the tick when current process is dispatched
                     dispatched_tick = cur_tick;
                 }
                 cur_process.cur_service_tick++; // increment the num of ticks that have been spent on current service
-                if (cur_process.cur_service_tick >= cur_process.cur_service.time_cost)
+                if (cur_process.cur_service_tick >= cur_process.getCurService().getServiceTime())
```
Comment (alibimm) [inline CS3343-Project/src/Project/FCFS.java]: getCurSurviceTime() instead of two?

File: CS3343-Project/src/Project/FCFS.java [hunk 3]
```diff
@@ -46,16 +56,16 @@ public Result schedule(ArrayList<Process> processes){
...
                 { // current service is completed
                     Manage_Next_Service_FCFS.manage_next_service_fcfs(cur_process, complete_num, dispatched_tick, cur_tick, ready_queue,
```
Comment (alibimm) [inline CS3343-Project/src/Project/FCFS.java]: Let's decide if we are gonna use the camel case of the underscore. Let's keep it consistent

File: CS3343-Project/src/Project/Manage_Next_Service_FCFS.java [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
+        boolean process_completed = cur_process.proceedToNextService();
         if (process_completed)
         { // the whole process is completed
             complete_num++;
-            cur_process.log_working(dispatched_tick, cur_tick + 1);
+            logger.logWorking(dispatched_tick, cur_tick + 1);
             System_helper.move_process_from(ready_queue, processes_done); // remove current process from ready queue
         }
-        else if (cur_process.cur_service.type == "K")
+        else if (cur_process.getCurService().getType() == ServiceType.Keyboard)
```
Comment (alibimm) [inline CS3343-Project/src/Project/Manage_Next_Service_FCFS.java]: one method call instead of two

**Review by alibimm — CHANGES_REQUESTED** [review]

**Pull request merged**